test/rules/invariant_booleans.dart

```diff
@@ -237,3 +237,13 @@ void bug372(bool foo) {
     // doSomethingElse();
   }
 }
+
+void bug337(int offset, int length) {
+  if (offset >= length) {
+    return;
+  }
+
+  offset++;
```

Comment:
Thanks, I was confused, somehow I thought this was the rule's code. Will add those cases to the test.